### PR TITLE
fix: make pubsub.unsubscribe async and alter pubsub.subscribe signature

### PR DESCRIPTION
**Pull request by alanshaw**
depends on https://github.com/ipfs/interface-ipfs-core/pull/260 and https://github.com/ipfs/js-ipfs-api/pull/760

### CODE REVIEW

File: src/core/components/pubsub.js [hunk 1]
```diff
@@ -20,13 +19,19 @@ module.exports = function pubsub (self) {
...
 
-    unsubscribe: (topic, handler) => {
+    unsubscribe: (topic, handler, callback) => {
       self._libp2pNode.pubsub.unsubscribe(topic, handler)
+
+      if (!callback) {
+        return Promise.resolve()
+      }
+
+      process.nextTick(() => callback())
```
Comment (diasdavid) [inline src/core/components/pubsub.js]: not all bundlers know how to handle process.nextTick well. We always use `async/setImmediate` for that matter.

**Comment by alanshaw** [comment]
@diasdavid tests not passing here because `ipfsd-ctl` depends on `ipfs-api@^20`. I'll submit a PR there now.

**Comment by alanshaw** [comment]
@diasdavid pubsub tests now pass for me when I npm link `ipfsd-ctl` on this branch https://github.com/ipfs/js-ipfsd-ctl/pull/245

**Comment by alanshaw** [comment]
🎉 only unrelated DNS test failed...

**Pull request merged**